RFC-0006-nan-definition.md

```diff
@@ -0,0 +1,179 @@
+
+### Other databases 
+
+#### Postgresql
+
+The NaN (not a number) value is used to represent undefined calculational results. In general, any operation with a NaN input yields another NaN. The only exception is when the operation's other inputs are such that the same output would be obtained if the NaN were to be replaced by any finite or infinite numeric value; then, that output value is used for NaN too. (An example of this principle is that NaN raised to the zero power yields one.)
+Note: In most implementations of the “not-a-number” concept, NaN is not considered equal to any other numeric value (including NaN). In order to allow numeric values to be sorted and used in tree-based indexes, PostgreSQL treats NaN values as equal, and greater than all non-NaN values.
+https://www.postgresql.org/docs/current/datatype-numeric.html#DATATYPE-NUMERIC-DECIMAL 
+
+#### Spark SQL
+
+There is special handling for not-a-number (NaN) when dealing with float or double types that does not exactly match standard floating point semantics. Specifically:
+NaN = NaN returns true.
+In aggregations, all NaN values are grouped together.
+NaN is treated as a normal value in join keys.
+NaN values go last when in ascending order, larger than any other numeric value. (this holds for min/max functions as well, e.g. array_max)
+ https://spark.apache.org/docs/3.0.0-preview/sql-ref-nan-semantics.html
+
+#### SnowFlake
+
+In snowflake, NaN=NaN and is sorted as greater than infinity: https://docs.snowflake.com/en/sql-reference/data-types-numeric#special-values 
+
+#### BigQuery
+
+NaN is considered smaller than -Infinity for sorting. 
+All Nan values are considered equal for grouping and distinct operations
+min/max functions return NaN if NaN is in the input
+Big query does not have functions that do distincts or sorting on arrays or maps
+https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types 
+https://cloud.google.com/bigquery/docs/reference/standard-sql/aggregate_functions#max
+
+#### MySQL
+
+Mysql doesn't support NaNs at all (see reference here https://dev.mysql.com/doc/refman/8.0/en/type-conversion.html)
+
+#### Oracle
+Oracle orders NaN greatest with respect to all other values, and evaluates NaN equal to NaN.
+
+https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/Data-Types.html#GUID-33A52FDB-BA5C-474E-96D3-40390BA5F5F4
+
+#### DB2
+
+DB2 has 2 kinds of floating point numbers.  The first, DOUBLE or FLOAT, does not support NaN or Infinity values.  The second, DECFLOAT, supports Infinities as well as quiet and signaling NaNs. It is described as an IEEE-754 number, but does not follow the IEEE spec for comparisons.  The NaNs compare as equal to each other and greater magnitude than any other number (The ordering among the different special values is as follows: -NAN < -SNAN < -INFINITY < 0 < INFINITY < SNAN <NAN)
```

Comment:
```suggestion
DB2 has 2 kinds of floating point numbers.  The first, DOUBLE or FLOAT, does not support NaN or Infinity values.  The second, DECFLOAT, supports Infinities as well as quiet and signaling NaNs. It is described as an IEEE-754 number, but does not follow the IEEE spec for comparisons.  The NaNs compare as equal to each other and greater in magnitude than any other number. The ordering among the different special values is as follows: -NAN < -SNAN < -INFINITY < 0 < INFINITY < SNAN <NAN.
```

This paragraph might also benefit from separating out "The first" and "The second" as left-justified, like the items in the Spark SQL an BigQuery topics.